nav2_regulated_pure_pursuit_controller/src/parameter_handler.cpp

```diff
@@ -162,56 +162,80 @@ ParameterHandler::ParameterHandler(
   node->get_parameter(
     plugin_name_ + ".max_robot_pose_search_dist",
     params_.max_robot_pose_search_dist);
-  if (params_.max_robot_pose_search_dist < 0.0) {
-    RCLCPP_WARN(
-      logger_, "Max robot search distance is negative, setting to max to search"
-      " every point on path for the closest value.");
-    params_.max_robot_pose_search_dist = std::numeric_limits<double>::max();
-  }
 
   node->get_parameter(
     plugin_name_ + ".interpolate_curvature_after_goal",
     params_.interpolate_curvature_after_goal);
-  if (!params_.use_fixed_curvature_lookahead && params_.interpolate_curvature_after_goal) {
-    RCLCPP_WARN(
-      logger_, "For interpolate_curvature_after_goal to be set to true, "
-      "use_fixed_curvature_lookahead should be true, it is currently set to false. Disabling.");
-    params_.interpolate_curvature_after_goal = false;
-  }
   node->get_parameter(
     plugin_name_ + ".use_collision_detection",
     params_.use_collision_detection);
 
-  if (params_.inflation_cost_scaling_factor <= 0.0) {
-    RCLCPP_WARN(
-      logger_, "The value inflation_cost_scaling_factor is incorrectly set, "
-      "it should be >0. Disabling cost regulated linear velocity scaling.");
-    params_.use_cost_regulated_linear_velocity_scaling = false;
-  }
 
-  dyn_params_handler_ = node->add_on_set_parameters_callback(
+  on_set_params_handler_ = node->add_on_set_parameters_callback(
+    std::bind(
+      &ParameterHandler::validateParameterUpdatesCallback,
+      this, std::placeholders::_1));
+  post_set_params_handler_ = node->add_post_set_parameters_callback(
+    std::bind(
+      &ParameterHandler::updateParametersCallback,
+      this, std::placeholders::_1));
+  pre_set_params_handler_ = node->add_pre_set_parameters_callback(
```

Comment:
Is the pre-set actually triggered on the initialization of the node to remove the checks above? If so, wouldn't we be able to remove all the parameter gets because the pre/post/on-set would be triggered on initialization?